handler/userHandler_test.go

```diff
@@ -0,0 +1,250 @@
+package handler
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/gin-gonic/gin"
+	"github.com/google/uuid"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler/dto/request"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler/dto/response"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/handler/mapper"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/mocks"
+	"github.com/murilogilfelpeto/ps-tag-onboarding-go/service/models"
+	"github.com/stretchr/testify/assert"
+	"io"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+)
+
+func TestSaveUser(t *testing.T) {
+	t.Run("Persist User", func(t *testing.T) {
+		gin.SetMode(gin.TestMode)
+		recorder := httptest.NewRecorder()
+		ctx, _ := gin.CreateTestContext(recorder)
+		mockService := &mocks.Service{}
+
+		requestBody := request.UserRequestDto{
+			FirstName: "John",
+			LastName:  "Doe",
+			Email:     "johndoe@email.com",
+			Age:       18,
+		}
+
+		jsonBody, _ := json.Marshal(requestBody)
+		ctx.Request = &http.Request{
+			Body: io.NopCloser(bytes.NewBuffer(jsonBody)),
+		}
+
+		user, _ := mapper.UserRequestToUser(requestBody)
+		createdUser, _ := models.NewUser("f7d2ea4b-a4d0-4103-9c63-55ec7977e4d1", "John", "Doe", "john.doe@email.com", 36)
+		mockService.On("SaveUser", ctx, user).Return(createdUser, nil)
+
+		handler := &Handler{
+			service: mockService,
+		}
+
+		handler.Save(ctx)
+		var responseBody response.UserResponseDto
+		_ = json.Unmarshal(recorder.Body.Bytes(), &responseBody)
+
+		assert.Equal(t, http.StatusCreated, recorder.Code)
+		assert.Equal(t, createdUser.GetID(), responseBody.ID)
+		assert.Equal(t, createdUser.GetFirstName(), responseBody.FirstName)
+		assert.Equal(t, createdUser.GetLastName(), responseBody.LastName)
+		assert.Equal(t, createdUser.GetEmail(), responseBody.Email)
+		assert.Equal(t, createdUser.GetAge(), responseBody.Age)
+	})
+
+	t.Run("Error Binding Json all fields", func(t *testing.T) {
+		gin.SetMode(gin.TestMode)
```

Comment:
Gin mode can be done once in the beginning of the test.

Reply:
my bad, in Java we have the habit to declare the testing settings inside the test to avoid unexpected behaviors. I'll fix this in all tests